Cargo.toml

```diff
@@ -22,6 +22,7 @@ edition = "2018"
 default = ["curl-client", "middleware-logger", "encoding"]
 curl-client = ["http-client/curl_client", "once_cell", "default-client"]
 h1-client = ["http-client/h1_client", "default-client"]
+h1-client-rustls = ["http-client/h1_client_rustls", "default-client"]
```

Comment:
I have a PR up on http-client, I'd like to separate the feature flag out there first even if we can't here: https://github.com/http-rs/http-client/pull/67

Reply:
So, once you merge it in http-client, would this PR still make sense ?
I can either adapt this PR to have:
```toml
h1-client = ["http-client/h1_client", "http-client/native-tls", "default-client"]
h1-client-rustls = ["http-client/h1_client", "http-client/rustls", "default-client"]
```

Or you can either re-expose the same `native-tls` and `rustls` in surf:
```toml
h1-client = ["http-client/h1_client", "default-client"]
native-tls = ["http-client/native-tls"]
rustls = ["http-client/rustls"]
```
Meaning that user wanting h1-client would have to use the combination of `h1-client` + `native-tls` or `rustls `.

What do you prefer ?

Reply:
You know what, since we only support this on h1 client anyways (due the lack of `rustls` support in `hyper-tls`), let's separate these and then whenever https://github.com/rust-lang/cargo/issues/8832 exists and `hyper-tls` supports `rustls` we can worry about that then, since this is the ideal path.

Reply:
Done:
 - Bump http-client to 6.3.0
 - Removed `h1-client-rustls` feature and added `native-tls` + `rustls`
 - Adapted ci.yaml since using `h1-client` feature only doesn't work anymore
 - Updated inline doc for features in src/lib.rs

Let me know if it's fine for you!